### PR TITLE
Filebeat 6.5

### PR DESCRIPTION
**Pull request by danielkasen**
Add correct gauges, add metric keys for filebeat 6.5, add additional offset metric to track uncompressed bytes

What does this PR do?
Adds the offset metric so you can track the uncompressed bytes using the registry file. Also handles rollover offset and 0 values. This also converts the increment to a counter since increment is deprecated for other metrics. Additionally, this adds the metrics names used for filebeat 6.5.

Motivation
Needed to find total log file size generated and using a newer version that had different keys in the filebeat stats api.

Review checklist
  PR has a meaningful title or PR has the no-changelog label attached
[None as it should use the current suite] Feature or bugfix has tests
  Git history is clean
  If PR impacts documentation, docs team has been notified or an issue has been opened on the documentation repo
Additional Notes
Anything else we should know when reviewing?

### CODE REVIEW

**Comment by danielkasen** [comment]
So I believe some of these test are failing due to the change in metric names. Any advice on how to proceed? Should I modify the names in the test file as well?

**Comment by therve** [comment]
Indeed at least some failures are due to the removal of some metrics, in particular the logstash and kafka ones. Are those not reported by filebeat anymore? If that's the case which version has made the change?

**Comment by l0k0ms** [comment]
Hello @danielkasen following up on https://github.com/DataDog/documentation/issues/4437, are you adding all of this metrics by default? If so you should update the `metadata.csv` accordingly following those guidelines: https://docs.datadoghq.com/developers/integrations/new_check_howto/#metrics-metadata-file

File: filebeat/datadog_checks/filebeat/filebeat.py [hunk 1]
```diff
@@ -28,59 +28,74 @@ class FilebeatCheckHttpProfiler:
...
+        'libbeat.config.module.stops'
+   ]
+
+    GAUGE_METRIC_NAMES = [
+        'filebeat.harvester.started',
+        'filebeat.harvester.open_files',
+        'filebeat.harvester.skipped',
+        'filebeat.harvester.running',
+        'filebeat.harvester.closed',
+        'filebeat.harvester.running'
```
Comment (l0k0ms) [inline filebeat/datadog_checks/filebeat/filebeat.py]: This Metric is defined twice (see two line above)

**Comment by l0k0ms** [comment]
Hello @danielkasen following up on https://github.com/DataDog/documentation/issues/4437, are you adding all of this metrics by default? If so you should update the `metadata.csv` accordingly following those guidelines: https://docs.datadoghq.com/developers/integrations/new_check_howto/#metrics-metadata-file

**Comment by l0k0ms** [comment]
@danielkasen I have prepared the new `metadata.csv` file according to the current list of metrics:

**Note**: I think we should prefix all metrics with `filebeat` though in order to make it easier to find all related metrics in the DD app. 

```
metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
filebeat.registry.unprocessed_bytes,gauge,,byte,,Number of bytes not processed by filebeat yet.,0,filebeat, fb reg unprocessed
beat.info.uptime.ms,count,,,,DESC,0,filebeat, fb info uptime
beat.memstats.rss,count,,,,DESC,0,filebeat, fb memstats rss
beat.memstats.gc_next,count,,,,DESC,0,filebeat, fb memstats gc
beat.memstats.memory_total,count,,,,DESC,0,filebeat, fb memstats total
beat.memstats.memory_alloc,count,,,,DESC,0,filebeat, fb memstats alloc
beat.handles.open,count,,,,DESC,0,filebeat, fb handles open
beat.handles.limit.hard,count,,,,DESC,0,filebeat, fb handles limit hard
beat.handles.limit.soft,count,,,,DESC,0,filebeat, fb handles limit soft
beat.cpu.total.ticks,count,,,,DESC,0,filebeat, fb cpu total ticks
beat.cpu.total.value,count,,,,DESC,0,filebeat, fb cpu total value
beat.cpu.total.time.ms,count,,,,DESC,0,filebeat, fb cpu total time
beat.cpu.system.ticks,count,,,,DESC,0,filebeat, fb sys ticks
beat.cpu.system.time.ms,count,,,,DESC,0,filebeat, fb sys user time
beat.cpu.user.ticks,count,,,,DESC,0,filebeat, fb user ticks
beat.cpu.user.time.ms,count,,,,DESC,0,filebeat, fb user time ms
filebeat.input.log.files.renamed,count,,,,DESC,0,filebeat, fb input log file
filebeat.input.log.files.truncated,count,,,,DESC,0,filebeat, fb input log file trunc
filebeat.events.active,count,,,,DESC,0,filebeat, fb event active
filebeat.events.added,count,,,,DESC,0,filebeat, fb event added
filebeat.events.done,count,,,,DESC,0,filebeat, fb event done
registrar.states.current,count,,,,DESC,0,filebeat, fb reg state current
registrar.states.cleanup,count,,,,DESC,0,filebeat, fb reg cleanup
registrar.states.update,count,,,,DESC,0,filebeat, fb reg update
registrar.writes.fail,count,,,,DESC,0,filebeat, fb write  fail
registrar.writes.total,count,,,,DESC,0,filebeat, fb write total
registrar.writes.success,count,,,,DESC,0,filebeat, fb write success
system.load.1,count,,,,DESC,0,filebeat, fb sys load 1
system.load.5,count,,,,DESC,0,filebeat, fb sys load 5
system.load.15,count,,,,DESC,0,filebeat, fb sys load 15 
system.load.norm.1,count,,,,DESC,0,filebeat, fb sys load norm 1
system.load.norm.5,count,,,,DESC,0,filebeat, fb sys load norm  5
system.load.norm.15,count,,,,DESC,0,filebeat, fb sys load norm 15
system.cpu.cores,count,,,,DESC,0,filebeat, fb sys cpu core
libbeat.output.read.errors,count,,,,DESC,0,filebeat, fb out read error 
libbeat.output.read.bytes,count,,,,DESC,0,filebeat, fb out read byte
libbeat.output.write.errors,count,,,,DESC,0,filebeat, fb out write error
libbeat.output.write.bytes,count,,,,DESC,0,filebeat, fb out write byte 
libbeat.output.events.batches,count,,,,DESC,0,filebeat, fbout event batches
libbeat.output.events.duplicates,count,,,,DESC,0,filebeat, fbout event duplicates
libbeat.output.events.acked,count,,,,DESC,0,filebeat, fbout event acked
libbeat.output.events.failed,count,,,,DESC,0,filebeat, fbout event failed
libbeat.output.events.dropped,count,,,,DESC,0,filebeat, fbout event dropped
libbeat.output.events.active,count,,,,DESC,0,filebeat, fb out event active
libbeat.output.events.total,count,,,,DESC,0,filebeat, fb out event total
libbeat.pipeline.queue.acked,count,,,,DESC,0,filebeat, fb pip queu
libbeat.pipeline.clients,count,,,,DESC,0,filebeat, fb pip client
libbeat.pipeline.events.retry,count,,,,DESC,0,filebeat, fb pip event retry
libbeat.pipeline.events.failed,count,,,,DESC,0,filebeat, fb pip event failed
libbeat.pipeline.events.dropped,count,,,,DESC,0,filebeat, fb pip event dropped
libbeat.pipeline.events.published,count,,,,DESC,0,filebeat, fb pip event published
libbeat.pipeline.events.active,count,,,,DESC,0,filebeat, fb pip event active
libbeat.pipeline.events.filtered,count,,,,DESC,0,filebeat, fb pip event filtered
libbeat.pipeline.events.total,count,,,,DESC,0,filebeat, fb pip event total
libbeat.config.reloads,count,,,,DESC,0,filebeat, fb cfg reload
libbeat.config.module.starts,count,,,,DESC,0,filebeat, fb cfg md start
libbeat.config.module.running,count,,,,DESC,0,filebeat, fb cfg md run
libbeat.config.module.stops,count,,,,DESC,0,filebeat, fb cfg md stop
filebeat.harvester.started,gauge,,,,DESC,0,filebeat,fb harv started
filebeat.harvester.open_files,gauge,,,,DESC,0,filebeat,fb harv open file 
filebeat.harvester.skipped,gauge,,,,DESC,0,filebeat,fb harv skipped
filebeat.harvester.running,gauge,,,,DESC,0,filebeat,fb harv runing 
filebeat.harvester.closed,gauge,,,,DESC,0,filebeat,fb harv closed
```

**Comment by therve** [comment]
Yeah I think it would make sense to have the prefix be filebeat everywhere. In particular the system ones look weird.

**Pull request closed without merging**